cli/src/samples.rs

```diff
@@ -103,3 +103,65 @@ pub fn get_config(trusted_peers: HashSet<PeerId>, key_pair: Option<KeyPair>) ->
     .build()
     .expect("Iroha config should build as all required fields were provided")
 }
+
+#[allow(clippy::implicit_hasher)]
+/// Get a sample Iroha configuration proxy. Trusted peers must either be
+/// specified in this function, including the current peer. Use [`get_trusted_peers`]
+/// to populate `trusted_peers` if in doubt. Almost equivalent to the [`get_config`]
+/// function, except the fact that the proxy is left unbuilt.
+///
+/// # Panics
+/// when [`KeyPair`] generation fails (rare).
+pub fn get_config_proxy(
+    trusted_peers: HashSet<PeerId>,
+    key_pair: Option<KeyPair>,
+) -> ConfigurationProxy {
+    let (public_key, private_key) = match key_pair {
+        Some(key_pair) => key_pair.into(),
+        None => KeyPair::generate()
+            .expect("Key pair generation failed")
+            .into(),
+    };
+    iroha_logger::info!(?public_key);
+    ConfigurationProxy {
+        public_key: Some(public_key.clone()),
+        private_key: Some(private_key.clone()),
+        sumeragi: Some(iroha_config::sumeragi::ConfigurationProxy {
+            key_pair: None,
+            peer_id: None,
+            trusted_peers: Some(TrustedPeers {
+                peers: trusted_peers,
+            }),
+            ..iroha_config::sumeragi::ConfigurationProxy::default()
+        }),
+        torii: Some(iroha_config::torii::ConfigurationProxy {
+            p2p_addr: Some(DEFAULT_TORII_P2P_ADDR.to_owned()),
```

Comment:
Why couldn't this entire struct be `DEFAULT_TORII_CONFIG_PROXY`?

Reply:
Because technically we're not having these addresses set currently in `torii`'s `Default` peer config, thus requiring them to be provided. So those constants are pretty much only used here and for client config.